ls1intum/Artemis
# [Exam] Quiz component for exams

### Checklist

### Motivation and Context


The current implementation of the quiz exercises for the different modes includes too many features which are not needed / shouldn't be available during an exam.

### Description

Extracted the base functionality of a quiz participation into a new component.
The participation is now an input of the component instead of being loaded via the url param.
The bar with the quiz-step icons is displayed on the left side and it can be navigated to the step by clicking it.

### Steps for Testing


**can only be tested locally until student exam participations aren't loaded into the exam view**

Setup:
1. Import and embed the component into a component which contains a quiz participation (e.g. the quiz component). You can also comment out the other component so that only the one to test is displayed.
2. Pass the participation into the quiz-exam-component.

Test:
1. Log in to Artemis
2. Create a quiz with every type of quiz question
3. Navigate to where the component is displayed (e.g. a quiz open for practice)
4. Check that the quiz' components are displayed correctly and that you can interact with them
5. Check that you can navigate to a specific quiz question via the bottom bar

### Screenshots

<img width="1424" alt="Bildschirmfoto 2020-06-18 um 15 48 47" src="https://user-images.githubusercontent.com/51987021/85028754-c5738f00-b17b-11ea-90d9-df9c2fc86d2b.png">
<img width="1421" alt="Bildschirmfoto 2020-06-18 um 15 48 36" src="https://user-images.githubusercontent.com/51987021/85028764-c9071600-b17b-11ea-9599-07935da503f3.png">
<img width="1430" alt="Bildschirmfoto 2020-06-18 um 15 48 28" src="https://user-images.githubusercontent.com/51987021/85028769-c99fac80-b17b-11ea-8a5b-d32cdda24c86.png">
